common/services/move.js

```diff
@@ -34,6 +49,43 @@ const moveService = {
     page = 1,
     isAggregation = false,
   } = {}) {
+    // TODO: This is more of a temporary solution to solve the problem where
+    // the API doesn't have a concept of what locations a user has access to
+    //
+    // Once Auth is moved to the API we would be able to remove this as the API
+    // would know to only return moves that a user has access to
+    if (
+      (filter['filter[from_location_id]'] || '').split(',').length >
+      LOCATIONS_BATCH_SIZE
+    ) {
+      return makeMultipleRequests(
+        moveService.getAll,
+        {
+          filter,
+          combinedData,
+          page,
+          isAggregation,
+        },
+        'filter["filter[from_location_id]"]'
+      )
+    }
+
+    if (
+      (filter['filter[to_location_id]'] || '').split(',').length >
+      LOCATIONS_BATCH_SIZE
+    ) {
+      return makeMultipleRequests(
+        moveService.getAll,
+        {
+          filter,
+          combinedData,
+          page,
+          isAggregation,
+        },
+        'filter["filter[to_location_id]"]'
+      )
+    }
+
```

Comment:
Might be useful adding a comment after this pointing out that this is where the real meat of the method begins

Reply:
I think your suggestion of moving the above chunks out into a separate method too might make that a bit clearer. Let me see if there's a tweak that can improve this.